### PR TITLE
Help panel implementation

### PR DESCRIPTION
**Pull request by Siddhi-agg**
Hey @panglesd , I thought of creating a new PR because of commit author issues in the previous #34 PR.

This PR aims to solve issue #11. I will continue my work from here on and keep pushing it for you to review.

### CODE REVIEW

**Comment by panglesd** [comment]
Sure! Next week I'll be off, so ping @Julow for reviews.

**Comment by Siddhi-agg** [comment]
Sure thing!

**Comment by Siddhi-agg** [comment]
@panglesd @Julow I have implemented the help panel and it works fine on my local setup. Please review my PR.

**Comment by Julow** [comment]
Unfortunately, there's now a conflict with the main branch. Don't hesitate to ask if you have a problem with this.

File: lib/interactive_viewer.ml [hunk 1]
```diff
@@ -62,14 +62,41 @@ let navigate z_patches (dir : direction) : unit =
...
+  let ui = 
+  Lwd.bind (Lwd.get help) ~f:(fun help_visible ->
+    match help_visible with
+    | true ->W.vbox
+    [
+      W.scrollbox @@ Lwd.pure @@ help_panel;
+      Lwd.pure @@ Ui.keyboard_area
+           (function
+             | `ASCII 'q', [] ->
+                 Lwd.set quit true;
```
[the remark this answers is not in the document]
Comment (Siddhi-agg) [reply lib/interactive_viewer.ml:90]: @Julow so do you want both ```q``` and ```h``` to quit the help panel or just ```q```` to quit and ```h``` to do nothing?

File: lib/interactive_viewer.ml [hunk 2]
```diff
@@ -86,10 +113,17 @@ let view (patches : Patch.t list) =
...
            (W.string
               "Type 'q' to quit, 'n' to go to the next operation, 'p' to go to \
-               the previous operation");
+               the previous operation, 'h' to go to the help panel");
     ]
-
+  ) in
+  W.vbox[
+   ui
+  ]
```
[the remark this answers is not in the document]
Comment (Siddhi-agg) [reply lib/interactive_viewer.ml:127]: @Julow Sorry for that. From now on, I will keep that in mind.

**Comment by Siddhi-agg** [comment]
> Unfortunately, there's now a conflict with the main branch. Don't hesitate to ask if you have a problem with this.

Since my changes are mostly in the view function and much hasn't been added to it, I think the conflict would not be very difficult to resolve. This is my first time dealing with conflicts, so I apologise if I am wrong.

File: lib/interactive_viewer.ml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Julow) [reply lib/interactive_viewer.ml:90]: I'd prefer `h` to do nothing in this case.

**Comment by Julow** [comment]
How the resolving of conflicts is going ?

**Comment by Siddhi-agg** [comment]
Hi @Julow , Is this the expected behaviour after successfully resolving merge conflicts. I am also wondering why my commits are being committed twice, In the start of the PR and now when I tried to resolve the conflicts.

**Comment by Julow** [comment]
It seems that you rebased the main branch over your branch, resulting in a slightly different version of the main branch. You then rebased your other commits on top of that.

You can fix this using interactive rebase:

Type `git rebase -i main`. This will open your editor with a list of commits, remove the commits that aren't from you.
Save and quit. Rebase should now have started.

Use `git status` to see in what state you are. Or even better, the VCS tab in vscode.
You might be in a conflict state: in this case, edit the files containing the conflict markers (starts with `<<<<<<<`).
When this is done, stage the files with `git add` (or from vscode) and do `git rebase --continue`.

This might find an other conflict.

**Comment by Siddhi-agg** [comment]
Hi, I am having a bit trouble pulling from the main branch, When I am trying to pull through rebase main, and then resolve conflict, My first commit is getting committed twice, and when I am trying to fetch changes through fetch command and then doing merge origin/main, All the files are getting added as if I am committing them. I have force pushed to my original commits without the other changes.

Can you please help me with this.

**Comment by Julow** [comment]
You can separate the pulling and the rebasing:
- Go to the main branch `git checkout main`
- Pull `git pull origin main` (if `origin` is not this repository, change it)
- Go back to your branch `git checkout patch-help-panel`
- Do the interactive rebase `git rebase -i main`.

**Comment by Julow** [comment]
An other tips: on your branch, squash the commits using `git rebase -i @~4` and changing all the commands to `squash` instead of `pick`, except for the first one.

**Comment by panglesd** [comment]
@Siddhi-agg let us know if you need more help with the rebase!

**Comment by Siddhi-agg** [comment]
@panglesd I tried to resolve conflicts the way you said. But I don't know why the ```<<<<<<<<<<<<HEAD``` is coming into my code. Also, I already had a ```zipper.ml``` in my branch and it got commited twice in the PR. Can you please tell me what I should do now?

**Comment by panglesd** [comment]
My suggestion would be the following, since the changes is not too big and the history not important:
- checkout the main branch, pull the latest changes, and start a new branch from there
- Use this PR's "change tab" to remember what were your changes, and adapt the changes to the latest change if needed.
- Commit in the new branch.
- Close this PR and open a new one.

Sorry, it is quite hard to help with git without knowing exactly what you did and what is your local state! But in any case, confidence with git rebases comes with experience!

**Comment by Siddhi-agg** [comment]
I see, I understood a lot of things through this PR, Especially about Git and resolving conflicts. Thanks for motivating me to further continue my work. I will open a new PR for the implementation.

**Pull request closed without merging**